ibis/client.py

```diff
@@ -982,6 +1000,180 @@ def _get_schema_using_query(self, query):
+                                       udf_info.output,
+                                       name, db)
+        self._execute(statement)
+
+    def create_uda(self, uda_info, uda_name=None, db=None):
+        """
+        Creates a user-defined aggregate function within Impala
+
+        Parameters
+        ----------
+        uda_info : UDAInfo object
+        uda_name : string (optional)
+        db : string (optional)
+        """
+        name = uda_info.get_name()
+        if uda_name:
+            name = uda_name
+        statement = ddl.CreateAggregateFunction(uda_info.hdfs_file,
+                                                uda_info.input_type,
+                                                uda_info.output_type,
+                                                uda_info.init_fn,
+                                                uda_info.update_fn,
+                                                uda_info.merge_fn,
+                                                uda_info.finalize_fn,
+                                                name,
+                                                db=db)
+        self._execute(statement)
+
+    def drop_udf(self, name=None, input_types=None, db=None, force=False,
+                 aggregate=False):
+        """
+        Drops a function
+        If only a database and force=True are passed, all functions
+        in specified database are dropped.
+        If only name is given, function will search for
+        Parameters
+        ----------
+        name : string
+        input_types : list of strings
+        must_exist : boolean, default False
+        db : string, default None
+        aggregate : boolean, default False
+        """
+        if not input_types:
+            if name:
+                #  TODO : Check if database is given.
+                #  If so, use fully qualified name to query
+                db = self.current_database()
+                result = self.list_udfs(db, like=name)
+                if len(result) > 1:
+                    if force:
+                        for func in result:
+                            self._drop_single_function(func.name, func.inputs,
+                                                       aggregate=aggregate)
+                        return
+                    else:
+                        raise Exception("More than one function " +
+                                        "with {0} found.".format(name) +
+                                        "Please specify force=True")
+                elif len(result) == 1:
+                    func = result.pop()
+                    self._drop_single_function(func.name, func.inputs,
+                                               aggregate=aggregate)
+                    return
+                else:
+                    raise Exception("No function found with name {0}"
+                                    .format(name))
+            elif db:
```

Comment:
I don't think this should be here. `drop_udf` should work like `drop_table` semantically (except for the dropping-multiple-overloaded-functions logic if no `input_types` passed)